### ISSUE TITLE
DataTable/TreeTable: initial sortBy without active-sort-icon and missing sortOrder param

### ISSUE THREAD

**Issue by edudoda** [primefaces/primefaces#7104]
**Describe the defect**

When dataTable or treeTable is initially sorted using a Collection (not SortMeta type) eg.: <p:dataTable sortBy="#{bean.xpto}" .../> it doesn't show the initial active-sorted-icon on that column and always defaults to descending order, it's only possible to change/use sortOrder="asc" when in each column (before we could set it as p:dataTable param).

And there is a strange behavior, in each column we set "sortOrder" param it becomes automatically initially sorted by default. 

**Reproducer**
https://github.com/edudoda/primefaces-test-cases

**Environment:**
 - PF Version: master-SNAPSHOT
 - JSF + version:  Mojarra 2.2.0
 - Affected browsers:  Chrome, iOS  

**To Reproduce**
1. Go to a page with  p:dataTable  or p:treeTable and sortBy param
2. Try to use the legacy code with  sortBy="" and  sortOrder=""
3.  Initial sort works but is always descending and user can't see it
3.1 If respective column has his own sortOrder="" it works as initial sort   bypassing  p:dataTable  settings

**Expected behavior**
1) We should can change initial sort order without need to use SortMeta class 
2) User should know which column and order it is already sorted
3) Setting a column sortOrder should not make it initially sorted (or should?), if so, sortBy from p:dataTable makes no sense

**Example XHTML**
```html
  <p:dataTable      value="#{bean.xpto}"      var="item"       
                             sortBy="#{item.op.test}"     >  
            <!--    sortBy=""  is always descending and we can't use  sortOrder="ascending"  like before  -->

        <p:column       sortBy="#{item.op.test}" 
                                 sortOrder="asc"      >  
              <!--    sortOrder="" makes this column initially sorted  so p:dataTable sortBy  is useless -->
				
                                  	 <f:facet name="header">header</f:facet>
					 <h:outputText value="#{item.op.test}"/>
	  </p:column>
 </p:dataTable>
																
```

**Comment by tandraschko** [comment]
@Rapster can you check this?

**Comment by Rapster** [comment]
Can't reproduce, please provide a reproducer. Regarding this point:

> Setting a column sortOrder should not make it initially sorted (or should?), if so, sortBy from p:dataTable makes no sense

Either use `DataTable#sortBy` or `Column#sortBy` using both doesn't make much sense IMO. So far, DataTable#sortBy has more priority than Column#sortBy, if you have a relevant use case I can think about changing it

**Comment by edudoda** [comment]
@Rapster I totally agree with you, but it's happening the opposite: `Column#sortBy` is overriding `DataTable#sortBy`.  Moreover, just with `Column#sortBy` we can get old behavior (PF8 and earlier): initial active-sorted-icon and change default order.

Commented [reproducer](https://github.com/edudoda/primefaces-test-cases) is attached now.

**Comment by Rapster** [comment]
Can't run it, I can see you modified few things in pom.xml... Don't have time for this, please provide a ready runnable reproducer with jetty (which primefaces-test is already configured for this)

**Comment by melloware** [comment]
@edudoda we should be able to just run `mvn clean jetty:run` and see your reproducer.

**Comment by edudoda** [comment]
OK just run it now using this command, please check reproducer again...

**Comment by Rapster** [comment]
Ok, please read migration guide here: https://primefaces.github.io/primefaces/10_0_0/#/../migrationguide/10_0_0 and breaking  changes. It contains every answers :wink: 

> DataTable#sortBy: no longer expect a Map, but a single or a collection of SortMeta (use SortMeta#builder) UIColumn#field or UIColumn#sortBy must be set to enable default sorting.

**Comment by edudoda** [comment]
I read it and still not understood. Can you clarify me this points:

1 - Like documentation says, I'm using a single on` DataTable#sortBy` and set `UIColumn#sortBy` and it has no effect to initially sort. (As you can see in first `p:dataTable` of reproducer)

2 -  `DataTable#sortOrder` was removed and replaced by `UIColumn#sortOrder` (defines default sort order), OK. But then,  now it's the only way I could get it initially sorted. So, the question is, for a single, when we  should  use ` DataTable#sortBy` if just using `UIColumn#` params we got things working? 

Thanks in advance

**Comment by melloware** [comment]
@edudoda it looks like you are passing `DataTable#sortBy` a "field" `sortBy="#{item.op.test}"` not a SortMeta?  Docs say it must be a  

> no longer expect a Map, but a single or a collection of SortMeta (use SortMeta#builder)

**Comment by edudoda** [comment]
@melloware  got it now. Yes, I was, because 'field' in columns (sort/filter) is still working and all my legacy code is using the old ''field'' logic.  As I didn't see any erros thrown, I was 'understanding/reading' that  "a single" still could be a field, not that should be a single of SortMeta type.

So, to reduce refactoring, better to me is switched to column#param approach and forget about new `DataTable#sortBy`.
In this case, in which the atribute is valid and his value don't, should we not see errors throws?
It will be easier to solve/remove any conflicts. _eg: SortBy value is not a SortMeta instance_

**Thank you for your time and considerations.**

**Comment by melloware** [comment]
How about your other ticket. Is that still an issue now that you have resolved how to refactor your code for 10.0.0?

**Comment by edudoda** [comment]
You mean #6987 ?    It's not related to default/initial sort stuff, but to the mechanism in which each column are sorted when user click in a sortAble column header.  If `UIColumn#sortBy` should still accept 'field', so yes, error persists when STATE_SAVING_METHOD = client. I just updated the [reproducer](https://github.com/edudoda/primefaces-test-cases) to reflect this.
